### PR TITLE
Add Makie.jl recipes

### PR DESCRIPTION
**Pull request by leonvonrabenmond**
Adds in the recipes for Makie.jl from #95.

### CODE REVIEW

File: src/plot-recipes.jl [hunk 1]
```diff
@@ -16,31 +16,58 @@
 ### Code:
 
 using RecipesBase
+using Makie
```
Comment (giordano) [inline src/plot-recipes.jl:19]: I'm definitely ***NOT*** going to accept a 100x increase in loading time.  `Makie` is a *massive* dependency.  Isn't there a better way to do this?  `Plots.jl` usess `RecipesBase.jl`, which is very lightweight.

**Review by giordano — CHANGES_REQUESTED** [review]

File: src/plot-recipes.jl (hunk 1 above)
Comment (leonvonrabenmond) [reply src/plot-recipes.jl:19]: I tried using MakieCore, but that only supports the PointBased plots. What could be done is to do the general conversion without the errorbar/band conversion explicitly 🤔 

I'll give that a try

**Comment by leonvonrabenmond** [comment]
So, this commit changes to `MakieCore`, which allows conversion of `scatter`, `lines` and similar (`PointBased` Plots). As far as I can tell, `errorbars` and `band` cannot be converted under `MakieCore` and require full `Makie`. 

I will check if it is possible to somehow achieve that.

File: src/plot-recipes.jl (hunk 1 above)
Comment (JaydevSR) [reply src/plot-recipes.jl]: Can you not try adding Makie as an optional dependency using `[extras]` in the `Project.toml`? You can see example of this done in Molly.jl where GLMakie is used this way (see files src/Molly.jl and ./Project.toml in the repo).

Comment (leonvonrabenmond) [reply src/plot-recipes.jl]: Yep, that is the solution I found as well by now and wanted to try. I just haven't found the time yet and, to be honest, forgot about it, so thank you very much.

The `[extras]` uses `Requires.jl` for exactly this purpose and that should totally work for us.

Comment (leonvonrabenmond) [reply src/plot-recipes.jl]: Using `Requires.jl` I now added all original functionality of this PR, while loading only should `Makie.jl` be loaded in the session. So we don't have the forever load time of `Makie`.

**Comment by alyaeanyx** [comment]
@giordano is there anything preventing this from being merged? This feature would be really nice to have while working with Makie.jl + Measurements.jl.

**Comment by giordano** [comment]
Making tests pass would be the very first thing.  I'll never merge a PR with failing tests.

File: src/makie-recipes.jl [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Makie.convert_arguments(P::Type{<:Errorbars}, x::AbstractVector{<:Measurement}, y::AbstractVector{<:Real}) =
+    convert_arguments(P, value.(x), y, uncertainty.(x))
+Makie.convert_arguments(P::Type{<:Errorbars}, x::AbstractVector{<:Real}, y::AbstractVector{<:Measurement}) =
+    convert_arguments(P, x, value.(y), uncertainty.(y))
+
+# band
+Makie.convert_arguments(P::Type{<:Band}, x::AbstractVector{<:Measurement}, y::AbstractVector{<:Measurement}) =
+    convert_arguments(P, value.(x), value.(y) - uncertainty.(y), value.(y) + uncertainty.(y))
+Makie.convert_arguments(P::Type{<:Band}, x::AbstractVector{<:Real}, y::AbstractVector{<:Measurement}) =
+    convert_arguments(P, x, value.(y) - uncertainty.(y), value.(y) + uncertainty.(y))
```
Comment (giordano) [inline src/makie-recipes.jl:23]: Missing new line
```suggestion
    convert_arguments(P, x, value.(y) - uncertainty.(y), value.(y) + uncertainty.(y))

```

File: src/plot-recipes.jl [hunk 3]
```diff
@@ -17,30 +17,32 @@
 
 using RecipesBase
 
-@recipe function f(y::AbstractArray{<:Measurement})
```
Comment (giordano) [inline src/plot-recipes.jl:20]: Why was this file changed at all?

File: README.md [hunk 4]
```diff
@@ -67,7 +67,8 @@ easy-to-use calculator.
   measurements with their physical units, perform numerical integration
   with [`QuadGK.jl`](https://github.com/JuliaMath/QuadGK.jl), numerical and
   automatic differentiation, and much more.
-* Integration with [`Plots.jl`](https://github.com/JuliaPlots/Plots.jl).
+* Integration with [`Plots.jl`](https://github.com/JuliaPlots/Plots.jl) and
```
Comment (giordano) [inline README.md:70]: Perhaps you'd want to mention this in the documentation as well: https://github.com/JuliaPhysics/Measurements.jl/blob/daa1ad282a02d945a1f6c69a8198998aa4b63002/docs/src/index.md#features-list

File: src/Measurements.jl [hunk 5]
```diff
@@ -126,4 +126,8 @@ include("show.jl")
 include("parsing.jl")
 include("plot-recipes.jl")
 
+function __init__()
```
Comment (giordano) [inline src/Measurements.jl:129]: There's already an `__init__` function.

**Comment by alyaeanyx** [comment]
> Making tests pass would be the very first thing. I'll never merge a PR with failing tests.
Oh, didn't notice that. Sorry.

**Comment by giordano** [comment]
I think this would be better implemented as a package extension: #133.

**Comment by giordano** [comment]
Closing in favour of #145
